File: core/src/main/java/io/github/manami/core/services/events/ProgressState.java
package io.github.manami.core.services.events;


/**
 * Used by services to notify observers of the progress.
 */
public class ProgressState {

  /**
   * Number of items which have already been processed.
   */
  private final int done;

  /**
   * Number of items which still need to be processed.
   */
  private final int todo;


  /**
   * Constructor
   *
   * @param done Number of processed items.
   * @param todo Number of items which still need to be processed.
   */
  public ProgressState(final int done, final int todo) {
    this.done = done;
    this.todo = todo;
  }

  public int getDone() {
    return done;
  }

  public int getTodo() {
    return todo;
  }
}

File: core/src/main/java/io/github/manami/core/tasks/events/ProgressState.kt
package io.github.manami.core.tasks.events


/**
 * Used by services to notify observers of the progress.
 * @param done Number of processed items.
 * @param todo Number of items which still need to be processed.
 */
open class ProgressState(
        /**
         * Number of items which have already been processed.
         */
        val done: Int,
        /**
         * Number of items which still need to be processed.
         */
        val todo: Int
)